peerprep-fe/src/app/collaboration/page.tsx

```diff
@@ -1,44 +1,52 @@
 'use client';
 
-import ProblemCodeEditor from '@/components/problems/ProblemCodeEditor';
 import ProblemDescriptionPanel from '@/components/problems/ProblemDescriptionPanel';
 import ProblemTable from '@/components/problems/ProblemTable';
-import { Button } from '@/components/ui/button';
-import {
-  Select,
-  SelectContent,
-  SelectItem,
-  SelectTrigger,
-  SelectValue,
-} from '@/components/ui/select';
 import { useFilteredProblems } from '@/hooks/useFilteredProblems';
-import { DEFAULT_CODE, SUPPORTED_PROGRAMMING_LANGUAGES } from '@/lib/constants';
+import { SUPPORTED_PROGRAMMING_LANGUAGES } from '@/lib/constants';
 import { Problem } from '@/types/types';
-import { UserCircle, UserX } from 'lucide-react';
-import React, { useCallback, useEffect, useRef, useState } from 'react';
+import React, {
+  Suspense,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { useSearchParams } from 'next/navigation';
-import * as Y from 'yjs';
-import { WebsocketProvider } from 'y-websocket';
-import { MonacoBinding } from 'y-monaco'
-import { editor as MonacoEditor } from 'monaco-editor';
-
-const CollaborationPage = () => {
-  const [selectionProblem, setSelectionProblem] = useState<Problem | null>(null);
+import dynamic from 'next/dynamic';
+import EditorSkeleton from './components/EditorSkeleton';
+import LoadingSpinner from '@/components/loading/LoadingSpinner';
+import { useCollaborationStore } from '@/state/useCollaborationStore';
+
+const CollaborationEditor = dynamic(
```

Comment:
not sure that dynamic does for this case, do we need it to lazy load?

Reply:
hmm reason was that if i didnt do this there were some build errors

Reply:
:o HAHA thats so weird